simpl/db/mongodb.py

```diff
@@ -0,0 +1,555 @@
+# Copyright (c) 2011-2013 Rackspace Hosting
+# All Rights Reserved.
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""MongoDB backend wrapper.
+
+This module exposes the SimplDB class which provides an opinoinated backend
+implementation that supports using MongoDB. The opinons of the implementation
+are:
+
+- we want to just supply a URL to connect to the database.
+- we want to use your own "id" fields to key documents.
+- we want the indexes to be specified in code.
+- want to perform basic CRUD operations on collections
+- we only want to simple, JSON objects (Jdicts, lists, strings and ints)
```

Comment:
- we only want to _use_ simple, JSON objects (Jdicts, lists, strings and ints) ?